go/vt/vtgate/planbuilder/jointree.go

```diff
@@ -34,6 +34,12 @@ type joinTree struct {
 	lhs, rhs queryTree
 
 	leftJoin bool
+
+	// predicatesToRemove lists all the predicates that needs to be removed
+	// from the right-hand side if we decide to do a hash join.
+	predicatesToRemove []sqlparser.Expr
```

Comment:
Should we rename this field to include Hash Join or something like that, it kinda makes me think that these are the predicates we are going to remove from this Join and not rhs? No strong opinions, maybe its just me who is weird

Reply:
agreed